native/test/src/tests/room/room_aliases.rs

```diff
@@ -0,0 +1,116 @@
+use acter::api::TimelineItem;
+use acter_core::{models::status::RoomAliasesContent, util::do_vecs_match};
+use anyhow::Result;
+use core::time::Duration;
+use futures::{pin_mut, stream::StreamExt, FutureExt};
+use tokio::time::sleep;
+use tokio_retry::{
+    strategy::{jitter, FibonacciBackoff},
+    Retry,
+};
+
+use crate::utils::random_user_with_random_convo;
+
+#[tokio::test]
+#[ignore = "test failed to resolve alias on other server (#friendlyname:server.name) :("]
+async fn test_room_aliases() -> Result<()> {
+    let _ = env_logger::try_init();
+
+    let (mut user, room_id) = random_user_with_random_convo("room_aliases").await?;
+    let state_sync = user.start_sync();
+    state_sync.await_has_synced_history().await?;
+
+    // wait for sync to catch up
+    let retry_strategy = FibonacciBackoff::from_millis(100).map(jitter).take(10);
+    let fetcher_client = user.clone();
+    let target_id = room_id.clone();
+    Retry::spawn(retry_strategy, move || {
+        let client = fetcher_client.clone();
+        let room_id = target_id.clone();
+        async move { client.convo(room_id.to_string()).await }
+    })
+    .await?;
+
+    let convo = user.convo(room_id.to_string()).await?;
+    let timeline = convo.timeline_stream();
+    let stream = timeline.messages_stream();
+    pin_mut!(stream);
+
+    let aliases = vec!["#friendlyname:server.name".to_owned()];
+    let room_aliases = serde_json::to_string(&aliases)?;
+    let aliases_event_id = convo.set_aliases(room_aliases).await?;
```

Comment:
@gnunicorn `server.name` is only placeholder and we have no server that resolve `#friendlyname`.
We should use `m1` dev server for that?
I don't know if github would allow for CI to send request to another machine.

Reply:
the ci servers have the servername `localhost`. IF you have the alias set there - which for a test we can do, but probably should pick a random name - then it should be able to resolved it, like `#random:localhost` should resolve.